arcflow/utils/bulk_import.py

```diff
@@ -128,8 +172,122 @@ def csv_bulk_import(csv_directory=None, load_type='ao', only_validate='false'):
                 }),
             }
         ).json()
+        file_import_report["results_status"] = import_job.get("status")
+        file_import_report["results_id"] = import_job.get("id")
+        file_import_report["results_uri"] = import_job.get("uri")
+        file_import_report["results_warnings"] = import_job.get("warnings")
+
+        bulk_import_report.append(file_import_report)
         print(json.dumps(import_job, indent=4))
+    return bulk_import_report
+
+def save_report(path, report_list, validate_only):
+    """Function to create and save reports for tracking bulk imports."""
+    current_datetime = datetime.now().strftime('%Y-%m-%d-%H%M%S')
+    action = "validate" if validate_only else "import"
+    suffix = current_datetime + "_" + action
+    report_file_name_stem = Path(path).name + "_report_" + suffix
+    report_text_file_name = report_file_name_stem + ".txt"
+    report_save_path = os.path.join(path, "reports")
+
+    if not os.path.exists(report_save_path):
+        os.makedirs(report_save_path)
+
+    txt_report_save_path = os.path.join(report_save_path, report_text_file_name)
+    with open(txt_report_save_path, 'w', encoding='utf-8') as report:
+        print("Import Job Info", file=report)
+        json.dump(report_list, report, indent=4)
+
+    report_csv_file_name = report_file_name_stem + ".csv"
+
+    fieldnames = ['identifier','ead_id','repo_id', 'rid','only_validate','type','resource_id','error','results_status','results_warnings','results_id','results_uri',"csv_issue_count"]
+    
+    csv_report_save_path = os.path.join(report_save_path, report_csv_file_name)
+    with open(csv_report_save_path, "w", newline="", encoding='utf-8') as csvfile:
+        writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+        writer.writeheader()
+        for row in report_list:
+            writer.writerow(row)
+
+def check_job_status(client, repo_id, job_id):
+    """Function to check whether a job has completed (and thus output files are ready)."""
+    while True:
+        job_status_response = client.get(f'/repositories/{repo_id}/jobs/{job_id}')
```

Comment:
For parameter naming consistency.
```suggestion
def check_job_status(asnake_client, repo_id, job_id):
    """Function to check whether a job has completed (and thus output files are ready)."""
    while True:
        job_status_response = asnake_client.get(f'/repositories/{repo_id}/jobs/{job_id}')
```